src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -862,9 +860,10 @@ private static Expression BindParameterFromBody(ParameterInfo parameter, bool al
                 }
             }
 
-            factoryContext.Metadata.Add(DefaultAcceptsMetadata);
             var isOptional = IsOptionalParameter(parameter);
 
+            factoryContext.Metadata.Add(new AcceptsMetadata(parameter.ParameterType, isOptional, new string[] {"application/json"}));
```

Comment:
This needs to be moved down after `factoryContext.AllowEmptyRequestBody` is set:

```suggestion
            factoryContext.Metadata.Add(new AcceptsMetadata(parameter.ParameterType, factoryContext.AllowEmptyRequestBody, new string[] {"application/json"}));
```

Reply:
Makes sense for the empty body. cleaned.